app/src/main/java/org/jellyfin/androidtv/util/apiclient/BaseItemUtils.kt

```diff
@@ -0,0 +1,224 @@
+	}
+	val episodeNumber = when {
+		baseItemType != BaseItemType.Episode -> indexNumber?.toString()
+		parentIndexNumber == 0 -> context.getString(R.string.lbl_special)
+		indexNumber != null -> when {
+			indexNumberEnd != null -> context.getString(R.string.lbl_episode_range, indexNumber, indexNumberEnd)
+			else -> context.getString(R.string.lbl_episode_number, indexNumber)
+		}
+		else -> null
+	}
+	val seasonEpisodeNumbers = listOfNotNull(seasonNumber, episodeNumber).joinToString(":")
+
+	val nameSeparator = when (baseItemType) {
+		BaseItemType.Episode -> " — "
+		else -> ". "
+	}
+
+	return listOfNotNull(seasonEpisodeNumbers, name)
+		.filter { it.isNotEmpty() }
+		.joinToString(nameSeparator)
+}
+
+fun BaseItemDto.getSubName(context: Context) = when (baseItemType) {
+	BaseItemType.Episode -> when {
+		locationType == LocationType.Virtual && name != null && premiereDate != null ->
+			context.getString(
+				R.string.lbl_name_date,
+				name,
+				TimeUtils.getFriendlyDate(context, TimeUtils.convertToLocalDate(premiereDate))
+			)
+		else -> name.orEmpty()
+	}
+	BaseItemType.Season -> when {
+		childCount != null && childCount > 0 -> when {
+			childCount > 1 -> context.getString(R.string.lbl_num_episodes, childCount)
+			else -> context.getString(R.string.lbl_one_episode)
+		}
+		else -> ""
+	}
+	BaseItemType.MusicAlbum -> when {
+		childCount != null && childCount > 0 -> when {
+			childCount > 1 -> context.getString(R.string.lbl_num_songs, childCount)
+			else -> context.getString(R.string.lbl_one_song)
+		}
+		else -> ""
+	}
+	BaseItemType.Audio -> name.orEmpty()
+	else -> officialRating.orEmpty()
+}
+
+fun BaseItemDto.getProgramUnknownChannelName() =
+	TvManager.getChannel(TvManager.getAllChannelsIndex(channelId)).name.orEmpty()
+
+fun BaseItemDto.getProgramSubText(context: Context) = buildString {
+	// Add the channel name if set
+	if (channelName != null) {
+		append(channelName)
+		append(" - ")
+	}
+
+	// Add the episode title if set
+	if (episodeTitle != null) {
+		append(episodeTitle)
+		append(" ")
+	}
+
+	val startTime = Calendar.getInstance()
+	startTime.time = TimeUtils.convertToLocalDate(startDate)
+
+	// If the start time is on a different day, add the date
+	if (startTime[Calendar.DAY_OF_YEAR] != Calendar.getInstance()[Calendar.DAY_OF_YEAR]) {
+		append(TimeUtils.getFriendlyDate(context, startTime.time))
+		append(" ")
+	}
+
+	// Add the start and end time
+	val dateFormat = DateFormat.getTimeFormat(TvApp.getApplication())
+	append(context.getString(
+		R.string.lbl_time_range,
+		dateFormat.format(startTime.time),
+		dateFormat.format(TimeUtils.convertToLocalDate(endDate))
+	))
+}
+
+fun BaseItemDto.getFirstPerson(searchedType: PersonType) =
+	people?.find { it.personType == searchedType }
+
+fun BaseItemDto.buildChapterItems(): List<ChapterItemInfo> {
+	val chapterItems = mutableListOf<ChapterItemInfo>()
+	val options = ImageOptions().apply {
+		imageType = ImageType.Chapter
+	}
+	chapters.mapIndexed { i, dto ->
```

Comment:
`return chapters.mapIndexed { ... }`
no need to add to a separate list

Reply:
This didn't work for me: the function expects a return type of `List<ChapterItemInfo>`, but `chapters` is a list of `ChapterInfoDto`.

Reply:
Yeah so you're mapping it to the new type, you'd need to change the body of the function so it returns the newly created chapter object

Reply:
Oh I see now--made the change.